### ISSUE TITLE
Define exit behavior of the selection tool 

### ISSUE THREAD

**Issue by PJansky** [flxzt/rnote#414]
**Description**
After using the selection tool, using the stylus anywhere on the canvas (besides the selected object) draws a stroke with the pen tool. This is quite useful to quickly return to drawing, but I find this behavior unintuitive, because there is no indication in the UI, that this will happen. In addition this behavior is different from Xournal++. In Xournal++, the first use of the pen after selecting an object doesn't draw anything, it just switches to the pen tool.  
This isn't really a bug, the question is more about, should the current behavior stay or is the behavior from Xournal++ better. Personally I like the Xournal++ behavior more, because of the UI feedback and because many users of this application probably come from Xournal++ like I did.

**To Reproduce**
1. Select something 
2. Use the pen to click on canvas to "exit" the selection tool and return to the pen tool 
Result: A stroke has been drawn where you clicked on the canvas

**Expected behavior**
No stroke created and pen tool selected like in Xournal++

**Comment by flxzt** [comment]
Yeah agree, most would expect the behaviour that switching back to the original pen takes an additional pen down->up cycle.

**Comment by Doublonmousse** [comment]
I think this can be refined a little bit. 

What I have in mind is more in line with what Microsoft Journal is doing : When an object is selected and the pen is used somewhere else, if the stroke is short, then it's cancelled. But if not (if you write a longer stroke), the stroke is preserved.

The logic is probably similar to what's done in rnote currently : the pen switches back immediately, but we check the length of any stroke that cancelled a selection and decide based on that to keep it or not to keep it.

**Comment by flxzt** [comment]
That sounds like a good idea!
